addons/web/static/src/views/view_button/view_button_hook.js

```diff
@@ -64,7 +64,10 @@ export function useViewButtons(model, ref, options = {}) {
                     enableButtons(getEl(), manuallyDisabledButtons, enableAction);
                     return;
                 }
-                const closeDialog = clickParams.close && env.closeDialog;
+                if (clickParams.special === "save") {
+                    await options.saveButtonClicked(clickParams);
```

Comment:
So before this commit it didn't save either? Or did it save thanks to the "save on leave" feature? If so, doesn't it also save if you click on a special="cancel" button?

Reply:
The record was created by the dialog, but not applied in the many2one field from the form view (it remained empty).
Now, when created, it is selected right after the dialog closed

Reply:
I still don't understand why you call this function here. I would have assumed that the fix would only be to set closeDialog to true if clickParams.special (what you do line 70)

Reply:
Maybe we should talk about this PR. I'm wondering if the problem isn't in `beforeExecuteActionButton` of form_controller instead (where we save, but where we totally ignore the props saveRecord)